tests/System.Buffers.Experimental.Tests/MemoryTests.cs

```diff
@@ -90,5 +90,163 @@ static void TestLifetime(OwnedBuffer<byte> owned)
                 var span = copyStoredForLater.Span;
             });
         }
+
```

Comment:
@nietras  there are tests that use unpinned arrays - `IsSliceOf_SubSpansShouldMatch_BasicArrays` for example